### PR TITLE
Sample loading

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
HI

Hopefully better handling of sample loading 
  - Snakecase to camelcase in mount_sample
  - Removed Pin check from loaded_sample_changed
  - Added bw argument to take_snapshot needed by automatic loop centring

Marcus

### CODE REVIEW

File: mxcube3/routes/scutils.py [hunk 1]
```diff
@@ -109,17 +113,17 @@ def mount_sample(beamline_setup_hwobj,
                     msg = "Manual centring used, waiting for" +\
                           " user to center sample"
                     log.warning(msg)
-                    dm.start_centring_method(dm.CENTRING_METHOD_MANUAL)
+                    dm.startCentringMethod(dm.MANUAL3CLICK_MODE)
```
Comment (meguiraun) [inline mxcube3/routes/scutils.py]: this does not exist in the `GenericDiffractometer` (snakecase yes )

File: mxcube3/routes/signals.py [hunk 2]
```diff
@@ -137,14 +137,13 @@ def loaded_sample_changed(sample):
         barcode = ''
 
     logging.getLogger("HWR").info('loaded sample changed now is: ' + address)
-
-    if isinstance(sample, Pin):
+ 
+    try:
```
Comment (meguiraun) [inline mxcube3/routes/signals.py]: why did you remove this? I think it could also be ~Container

Comment (marcus-oscarsson) [reply mxcube3/routes/signals.py]: The code as it were before generated errors when executed, since sample_data was undefined if
sample was not a Pin

And I dont understand why it would be container, the event is sample changed, so sample should 
not be a container. For me it indicates the the event is called when its not supposed to be called

If its the case it should just return at the top of the function, because all the logic in this function corresponds to a sample beeing loaded and is not valid for a container.

File: mxcube3/routes/scutils.py (hunk 1 above)
Comment (marcus-oscarsson) [reply mxcube3/routes/scutils.py]: Yes I noticed, its unfortunately what we have on our beamlines so there is not much we can do about it for the moment. Both the Minidfiff and Microdiff classes uses the camel case version, which one are you using ?

Comment (meguiraun) [reply mxcube3/routes/scutils.py]: we are using the `snake_case`. I am fine with a `hasattr`, like we have done before in similar cases.

Comment (marcus-oscarsson) [reply mxcube3/routes/scutils.py]: This could get quite problematic, since for instance camel case is used in the queue_entry and perhaps in other places. The interface of GenericDiffractomter is not really applied, which is 
a problem in itself. This is I believe is due to that the work with it is not complete yet.

I noticed that you already have some camel case aliases in your diffractometer so Id suggest that you add one for startCentringMethod aswell so that its backward compatible with the rest of the queue.

We will continue the discussion with the new interface for hardware objects 3.0

Marcus

File: mxcube3/routes/signals.py (hunk 2 above)
Comment (meguiraun) [reply mxcube3/routes/signals.py]: Maybe I am mixing containers/baskets/pins etc...

This code was working with mockups and the Pin was needed (for some error I cannot remember), but I see and agree with your point, so we can put everything on the same `if`

File: mxcube3/routes/scutils.py (hunk 1 above)
Comment (meguiraun) [reply mxcube3/routes/scutils.py]: I guess I can do that

File: mxcube3/routes/signals.py (hunk 2 above)
Comment (meguiraun) [reply mxcube3/routes/signals.py]: this is a merge problem, the `try` should be inside the `if`
